pkg/controller/controller.go

```diff
@@ -341,6 +341,15 @@ func (ctrl *resizeController) pvNeedResize(pvc *v1.PersistentVolumeClaim, pv *v1
 		return false
 	}
 
+	// lets make sure that we are reading latest PV object so as any PV updates that might have happened
```

Comment:
What happens if you call resize twice on the same PV?

Reply:
in general once kubelet finishes resizing then all resizing related conditions are removed from PVC. But what is happening in e2e is - we take down the pod and then we expand the PVC and we wait for "FileSystemResizePending" condition on PVC, but if resize is called again in resize controller then that condition might get replaced by "Resizing" condition and hence e2e may flake.

Reply:
Does that mean we should keep an internal cache and update it after we make an API update, similar to what we do in pv controller? That way, we don't need to do a GET request on every loop, which will limit our scalibility.

Reply:
Also the reason I think this will eliminate this kind of flakes is because - the previous operation on PVC can't finish without writing the object to etcd. The only reason we would return true from here is if informers haven't been updated yet, but if we did read from apiserver directly, this problem should go away.

Alternatively we will have to maintain local cache like provisioner/snapshotter etc.

Reply:
I was thinking about local caching but I am not sure if it is worth doing in this case. The PV size comparison is always done *after* PVC size comparsion, so in this case PVC does have a volume expansion pending (like we know `pvc.Status.Capacity < pvc.Spec.Capacity`). So this API call should only be made twice per PVC size change.

Reply:
cc @RaunakShah. Can you run your benchmark tests against this PR so we can see the impact of adding this GET call?

Reply:
Sure!

Reply:
Spawning 10 threads to resize PVCs in a loop for 120s. Each thread moves onto the next resize operation once the PVC enters `FileSystemResizePending` condition. Dummy vSphere CSI driver that returns success for `ControllerExpandVolume` calls. 
- Without this change, using image `quay.io/k8scsi/csi-resizer:v1.0.0-rc2`:
Ran 134 operations with throughput of 1.127 ops/sec
- With this change:
Ran 88 operations with a throughput of 0.79 ops/sec

Reply:
So if I understand correctly, this PR overall reduced number of API requests from external-resizer?

Reply:
Quite the opposite unfortunately :) With the rc image, i can get 134 calls to `ControllerExpandVolume` but that drops to 88 with this PR, the reason being that each `ControllerExpandVolume` request takes longer to process.

Reply:
@jsafrane pointed out that we could do same thing @RaunakShah  did in provisioner - https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/pull/83 .  lets see if this helps.